### DISCUSSION TITLE
Change the default terminal emulator

### DISCUSSION THREAD

**Discussion by u-Mear** [ivan-hc/AM#16] [Ideas]
Hello,
I recently downloaded "AM" and really loved is a pretty piece that was missing in my machine now I can control all of my packages for arch, AUR, flatpak and now AppImages, that is really awesome.

and I noticed that he opens "xterm" to show the progress bar and that is a unique thing and it is pretty but personally I didn't really like it with all the windows on the screen and I see if would be a way to change the terminal or even disable this feature, btw I could change the terminal to kitty from "/opt/am/APP-MANAGER" but I don't really like to manipulate with this files because I do distro hopping a lot and I am not stuck with a specific distro for now until I complete my configurations to run a prefer "wm" I like so I will really happy if that feature would be there to disable it or change the terminal.

**Comment by ivan-hc** [comment]
hi, I use xterm at the line 326 of this script because the output must be clear on the default terminal emulator. Xterm is the only terminal emulator that is compatible with all WM and compositors, this is why I've included it in the dependences. I'm still looking for a better solution. I've also used `less -E` but this breaks the interactivity with the shell when you install something that requires a prompt (for example Firefox, requires to enter the language, or Libreoffice, to choose a version). For now I've not found a best solution. Sorry

**Reply by u-Mear** [reply]
thank you for responding and I really appreciate your work, take your time with it because it is not a real problem after all and I am okay with it. and thanks for all of your work.

**Reply by DLopezJr** [reply]
Why does the output need to be cleared?

**Reply by ivan-hc** [reply]
purely as a matter of aesthetics, nothing more

**Comment by ivan-hc** [comment]
@DLopezJr have you got an idea on how to replace xterm? This may also be useful in case we want to create a GUI as suggested in #8

**Reply by DLopezJr** [reply]
Could a select loop work for this? Something else to check out is to use a fuzzy finder. The main ones users have are fzf, dmenu, and rofi.

**Comment by ivan-hc** [comment]
here are 3 different approuches to install the apps: xterm (1, the default one, allows to interact when prompted questions), less (2, clean but non interactive) or nothing (not clean)

https://github.com/ivan-hc/AM-Application-Manager/assets/88724353/5dae2883-ac74-4806-97c5-aa9564859641

I can simply use `sudo ./$arg | less -E`(second example), but we will lost the prompting on the following installation scripts:

- firefox
- firefox-beta
- firefox-dev
- firefox-esr
- firefox-nightly
- libreoffice
- thunderbird
- thunderbird-beta

The prompt consists in choosing a version of Libreoffice or choosing a language for Firefox and Thunderbird.

This is the only reason I've implemented `xterm`, and this can be a stupid reason being less scripts if compared to the whole database.

Now I have an alternative solution:

- We can also change the installation scripts for Firefox/Thunderbird by including a function in `wget` to detect the user agent, so we can download that version for that language;
- Libreoffice... the prompt appear, but when you choose a number you can't read what you type. I can also remove that installation script, but the other Libreoffice-related scripts have a longer name to type (if you have not `bash-completion` installed) and the user does not know the difference between a version or another one.

What you suggest?

**Comment by ivan-hc** [comment]
OK, everything has been solved, I've edited the installation scripts for Firefox and Thunderbird in a way that they can automatically detect your language, so the archive will be downloaded for your language (if it is in the list of Mozilla).

I'm ready to remove xterm by replacing it with less, stay tuned!
